### PR TITLE
Authentication Backend

### PR DESCRIPTION
**Pull request by aShahriarSazim**
I created the auth module and defined sign In sign Up functionalities. I also defined the passport-jwt strategy which I'll use later for authorization/middleware/guards

### CODE REVIEW

File: server/src/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import {signInDto, signUpDto} from "./dto";
+import {AuthService} from "./auth.service";
+import {AuthGuard} from "@nestjs/passport";
+
+@Controller('auth')
+export class AuthController {
+
+    constructor(private AuthService: AuthService) {}
+
+    @Post('/local/signin')
```
Comment (rChowdhurySazim) [inline server/src/auth/auth.controller.ts:11]: should be 'siginin'  only. applies for everywhere you used local.

File: server/src/auth/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    async signInLocal(@Body() dto: signInDto) {
+        return this.AuthService.signInLocal(dto);
+    }
+    @Post ('/local/signup')
+    async signUpLocal(@Body() dto: signUpDto) {
+        return this.AuthService.signUpLocal(dto);
```
Comment (rChowdhurySazim) [inline server/src/auth/auth.controller.ts:17]: remove local

File: server/src/auth/auth.module.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import { JwtModule } from '@nestjs/jwt';
+import { AuthService } from './auth.service';
+import {JwtStrategy} from "./strategy";
+
+@Module({
+  imports: [JwtModule.register({})],
+  controllers: [AuthController],
+  providers: [AuthService, JwtStrategy]
+})
+export class AuthModule {}
```
Comment (rChowdhurySazim) [inline server/src/auth/auth.module.ts:12]: Please add a newline at the end. Every file should have an empty newline at the end.

File: server/src/auth/dto/auth.dto.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    phone: string;
+
+    @IsString()
+    @IsNotEmpty()
+    address: string;
+
+    @IsString()
+    @IsNotEmpty()
+    password: string;
+}
```
Comment (rChowdhurySazim) [inline server/src/auth/dto/auth.dto.ts:37]: Please add a newline at the end. Every file should have an empty newline at the end.

File: server/src/auth/dto/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export * from './auth.dto';
```
Comment (rChowdhurySazim) [inline server/src/auth/dto/index.ts:1]: Please add a newline at the end. Every file should have an empty newline at the end.

File: server/src/auth/strategy/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export * from './jwt.strategy';
```
Comment (rChowdhurySazim) [inline server/src/auth/strategy/index.ts:1]: Please add a newline at the end. Every file should have an empty newline at the end.

File: server/src/auth/strategy/jwt.strategy.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        super({
+            jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+            ignoreExpiration: false,
+            secretOrKey: config.get('JWT_SECRET'),
+        });
+    }
+    async validate(payload: any) {
+        return {userId: payload.userId, email: payload.email};
+    }
+}
```
Comment (rChowdhurySazim) [inline server/src/auth/strategy/jwt.strategy.ts:19]: Please add a newline at the end. Every file should have an empty newline at the end.

File: server/src/prisma/prisma.module.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { Global, Module } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+
+@Global()
+@Module({
+  providers: [PrismaService],
+  exports: [PrismaService]
+})
+export class PrismaModule {}
```
Comment (rChowdhurySazim) [inline server/src/prisma/prisma.module.ts:9]: Please add a newline at the end. Every file should have an empty newline at the end.

File: server/src/prisma/prisma.service.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    constructor(config: ConfigService) {
+        super({
+            datasources: {
+                db: {
+                    url: config.get('DATABASE_URL')
+                }
+            }
+        });
+    }
+}
```
Comment (rChowdhurySazim) [inline server/src/prisma/prisma.service.ts:16]: Please add a newline at the end. Every file should have an empty newline at the end.

**Review by rChowdhurySazim — CHANGES_REQUESTED** [review]

File: server/.env.example [hunk 10]
```diff
@@ -4,4 +4,5 @@
 # Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB and CockroachDB.
 # See the documentation for all the connection string options: https://pris.ly/d/connection-strings
 
-DATABASE_URL="postgresql://user:pass@localhost:port/teebay-for-bcdg?schema=public"
\ No newline at end of file
+DATABASE_URL="postgresql://user:pass@localhost:port/teebay-for-bcdg?schema=public"
+JWT_SECRET="SOME RANDOM SECRET MAKE IT HARDER TO GUESS"
```
[the remark this answers is not in the document]
Comment (aShahriarSazim) [reply server/.env.example:8]: Done

File: server/src/auth/auth.controller.ts (hunk 1 above)
Comment (aShahriarSazim) [reply server/src/auth/auth.controller.ts:11]: Done

File: server/src/auth/auth.controller.ts (hunk 2 above)
Comment (aShahriarSazim) [reply server/src/auth/auth.controller.ts:17]: Done

File: server/src/auth/auth.module.ts (hunk 3 above)
Comment (aShahriarSazim) [reply server/src/auth/auth.module.ts:12]: Done

File: server/src/auth/dto/auth.dto.ts (hunk 4 above)
Comment (aShahriarSazim) [reply server/src/auth/dto/auth.dto.ts:37]: Done

File: server/src/auth/dto/index.ts (hunk 5 above)
Comment (aShahriarSazim) [reply server/src/auth/dto/index.ts:1]: Done

File: server/src/auth/strategy/index.ts (hunk 6 above)
Comment (aShahriarSazim) [reply server/src/auth/strategy/index.ts:1]: Done

File: server/src/auth/strategy/jwt.strategy.ts (hunk 7 above)
Comment (aShahriarSazim) [reply server/src/auth/strategy/jwt.strategy.ts:19]: Done

File: server/src/prisma/prisma.module.ts (hunk 8 above)
Comment (aShahriarSazim) [reply server/src/prisma/prisma.module.ts:9]: Done

File: server/src/auth/auth.controller.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+import {signInDto, signUpDto} from "./dto";
+import {AuthService} from "./auth.service";
+
+@Controller('auth')
+export class AuthController {
+
+    constructor(private AuthService: AuthService) {}
+
+    @Post('/signin')
+    async signInLocal(@Body() dto: signInDto) {
```
Comment (rChowdhurySazim) [inline server/src/auth/auth.controller.ts:11]: please remove local from the function name aswell

File: server/src/auth/auth.controller.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+        return this.AuthService.signIn(dto);
+    }
+
+    @Post ('/signup')
+    async signUpLocal(@Body() dto: signUpDto) {
```
Comment (rChowdhurySazim) [inline server/src/auth/auth.controller.ts:16]: please remove local from the function name aswell

**Review by rChowdhurySazim — CHANGES_REQUESTED** [review]

**Review by rChowdhurySazim — APPROVED** [review]

**Pull request closed without merging**